src/figanos/matplotlib/plot.py

```diff
@@ -1356,8 +1356,24 @@ def heatmap(
         raise TypeError("`data` must contain a xr.DataArray or xr.Dataset")
 
     # setup fig, axis
-    if ax is None:
+    if ax is None and ("row" not in plot_kw.keys() and "col" not in plot_kw.keys()):
         fig, ax = plt.subplots(**fig_kw)
+    elif ax is not None and ("col" in plot_kw or "row" in plot_kw):
+        raise ValueError("Cannot use 'ax' and 'col'/'row' at the same time.")
+    elif ax is None:
+        if any([k != "figsize" for k in fig_kw.keys()]):
+            warnings.warn(
+                "Only figsize arguments can be passed to fig_kw when using facetgrid."
+            )
+        plot_kw.setdefault("col", None)
+        plot_kw.setdefault("row", None)
+        heatmap_dims = list(
+            set(da.dims)
+            - {d for d in [plot_kw["col"], plot_kw["row"]] if d is not None}
+        )
+        if da.name is None:
+            da = da.to_dataset(name="data").data
+        da_name = da.name
```

Comment:
This part is a bit more rudimentary than in `gridmap` for instance, where `figure.add_subplot()` is also used in `fig_kw`
> "Only figsize and figure.add_subplot() arguments can be passed to fig_kw when using facetgrid."
I don't think I've seen an example on how to properly use `figure.add_subplot()` in this context, so I prefer not to advertise any of this. If this is a crucial feature I can try to incorporate it too.
I'm not sure how

Reply:
No i don't think it is crucial to have it. I was just wondering how much access we have to seaborn facetgrid options via the function heatmap